src/color.rs

```diff
@@ -0,0 +1,35 @@
+#![allow(dead_code)]
+
+pub const FG_BLACK: u32     = 30;
+pub const FG_RED: u32       = 31;
+pub const FG_GREEN: u32     = 32;
+pub const FG_YELLOW: u32    = 33;
+pub const FG_BLUE: u32      = 34;
+pub const FG_MAGENTA: u32   = 35;
+pub const FG_CYAN: u32      = 36;
+pub const FG_WHITE: u32     = 37;
+
+pub const BG_BLACK: u32     = 40;
+pub const BG_RED: u32       = 41;
+pub const BG_GREEN: u32     = 42;
+pub const BG_YELLOW: u32    = 43;
+pub const BG_BLUE: u32      = 44;
+pub const BG_MAGENTA: u32   = 45;
+pub const BG_CYAN: u32      = 46;
+pub const BG_WHITE: u32     = 47;
```

Comment:
Make a Color enum and add these as elements, then add a `as_background_color(self) -> u32` and `as_foreground_color(self) -> u32` functions.